File: InterfaceBuilding/SwiftInterfaceBulding/SwiftInterfaceBulding/NoChangeTester.swift
//
//  NoChangeTester.swift
//  SwiftInterfaceBulding
//
//  Created by Krzysztof Drab on 30/12/2020.
//  Copyright © 2020 Krzysztof Drab. All rights reserved.
//

import Foundation

class NoChangeTester: Tester {
    private var nbOfItems: Int = 0
    private var rows: [RowData] = []
    
    override func generateRows() -> [RowData] {
        self.tests = [1, 250, 500, 750, 1000, 1250, 1500, 1750, 2000]
        self.nbOfItems = tests[testId];
        self.rows = self.generateDefaultRows(nbOfRows: self.nbOfItems);
        return self.rows;
    }

    
    override func updateRows() -> [RowData] {
//        let startingPoint = Date()
//        var itemsToChange = 5;
//        while(itemsToChange > 0) {
//            let i = Int.random(in: 0..<self.rows.count);
//            let j = Int.random(in: 0..<self.rows.count);
//            rows.swapAt(i, j);
//            itemsToChange -= 1;
//        }
//        self.timeToModify += startingPoint.timeIntervalSinceNow * -1;
        return self.rows;
    }
    
    override func testEnded(fps: Double) {
        self.results.append(TestResult(n: nbOfItems, timeToModify: timeToModify, fps: fps))
    }
}

File: InterfaceBuilding/flutter_interface_building/lib/no_change_tester.dart
import 'dart:io' show Platform;
import 'dart:math';

import 'row_data.dart';
import 'tester.dart';

class NoChangeTester extends Tester {
  List<int> tests = Platform.isIOS
    ? [1, 250, 500, 750, 1000, 1250, 1500, 1750, 2000]
    : [1, 50, 100, 150, 200, 250, 300, 350, 400];
  int nbOfItems = 0;
  List<RowData> rows;

  List<RowData> generateRows() {
    this.nbOfItems = this.tests[this.testId];
    this.rows = this.generateDefaultRows(this.nbOfItems);
    return this.rows;
  }

  List<RowData> updateRows() {
    // final stopwatch = Stopwatch()..start();
    // var rng = new Random();
    // RowData temp;
    // for(var i = 0 ; i < 1 ; i++) {
    //   int a = rng.nextInt(this.nbOfItems);
    //   int b = rng.nextInt(this.nbOfItems);
    //   temp = this.rows[a];
    //   this.rows[a] = this.rows[b];
    //   this.rows[b] = temp;
    // }
    // this.timeToModify += stopwatch.elapsedMilliseconds;
    return this.rows;
  }

  void testEnded(double fps) {
    this.results.add(TestResult(this.nbOfItems, fps, this.timeToModify));
  }
}